### PR TITLE
[DM] IAM Member binding refactor to support on resource binding

### PR DESCRIPTION
**Pull request by ocsig**
Supporting `gcp-type:*:virtual.*.iamMemberBinding` on resource level. ( Currently implementing GCS, but this refactoring enables the easy addition of new virtual types.)

### CODE REVIEW

**Comment by ocsig** [comment]
FIX: #462

File: dm/templates/iam_member/iam_member.py [hunk 1]
```diff
@@ -15,71 +15,75 @@
...
-                iam_resource = {
-                    'name': (resourse_name),
-                    # TODO - Virtual type documentation needed
-                    'type': 'gcp-types/cloudresourcemanager-v1:virtual.projects.iamMemberBinding',
-                    'properties': {
-                        'resource': project_id,
-                        'role': role['role'],
-                        'member': member,
-                    }
+            resourse_name = '{}-{}'.format(policy_get_name,
```
Comment (sjvanrossum) [inline dm/templates/iam_member/iam_member.py]:
```suggestion
            resource_name = '{}-{}'.format(policy_get_name,
```

File: dm/templates/iam_member/iam_member.py [hunk 2]
```diff
@@ -15,71 +15,75 @@
...
+                                           base_resource['postfix'])
+            iam_resource = {
+                'name': resourse_name,
```
Comment (sjvanrossum) [inline dm/templates/iam_member/iam_member.py]:
```suggestion
                'name': resource_name,
```

File: dm/templates/iam_member/iam_member.py [hunk 3]
```diff
@@ -15,71 +15,75 @@
...
                 }
-                iam_resource.update(dependson)
-                resources.append(iam_resource)
+            }
+            iam_resource.update(dependson)
+            resources.append(iam_resource)
 
-            dependson = { 'metadata': { 'dependsOn': [resourse_name] + dependson_root } }
+            dependson = {'metadata': {'dependsOn': [
+                resourse_name] + dependson_root}}
```
Comment (sjvanrossum) [inline dm/templates/iam_member/iam_member.py]:
```suggestion
                resource_name] + dependson_root}}
```

**Pull request merged**